### PR TITLE
feat(FluentProvider): emit errors on duplicate IDs

### PR DESCRIPTION
**Pull request by layershifter**
This PR:
- updates docs to include a section about React 18
- `FluentProvider` emits errors on if proper configuration is missing and IDs are duplicated
  image

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28670/perf-test-react-components/InfoButton-mount_base.html">12</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28670/perf-test-react-components/InfoButton-mount.html">9</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28670/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28670/perf-test-react-components/Avatar-mount_base.html">625</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28670/perf-test-react-components/Avatar-mount.html">592</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28670/perf-test-react-components/Button-mount_base.html">293</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28670/perf-test-react-components/Button-mount.html">287</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28670/perf-test-react-components/Field-mount_base.html">1085</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28670/perf-test-react-components/Field-mount.html">1096</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28670/perf-test-react-components/FluentProvider-mount_base.html">661</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28670/perf-test-react-components/FluentProvider-mount.html">637</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28670/perf-test-react-components/FluentProviderWithTheme-mount_base.html">78</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28670/perf-test-react-components/FluentProviderWithTheme-mount.html">73</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28670/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">71</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28670/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">66</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28670/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">74</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28670/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">82</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28670/perf-test-react-components/InfoButton-mount_base.html">12</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28670/perf-test-react-components/InfoButton-mount.html">9</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28670/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28670/perf-test-react-components/MakeStyles-mount_base.html">846</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28670/perf-test-react-components/MakeStyles-mount.html">831</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28670/perf-test-react-components/Persona-mount_base.html">1614</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28670/perf-test-react-components/Persona-mount.html">1654</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28670/perf-test-react-components/SpinButton-mount_base.html">1299</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28670/perf-test-react-components/SpinButton-mount.html">1372</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                                                                             |      Baseline (minified/GZIP) |                           PR |                                                                                                                                                                                                  Change |
| :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: | ---------------------------: | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `67.576 kB`<br />`18.225 kB` | `67.544 kB`<br />`18.216 kB` | `-32 B` <br />`-9 B`  |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `208.062 kB`<br />`57.968 kB` |  `208.03 kB`<br />`57.96 kB` | `-32 B` <br />`-8 B`  |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `36.409 kB`<br />`12.003 kB` |  `36.409 kB`<br />`12.01 kB` |                                                                                                   <br />`7 B`  |
| <samp>react-provider</samp> <br /> <abbr title='bundle-size/FluentProvider.fixture.js'>FluentProvider</abbr>                                                                  |    `18.129 kB`<br />`6.73 kB` |  `18.129 kB`<br />`6.742 kB` |                                                                                                  <br />`12 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                     |      Size (minified/GZIP) |
| --------------------------------------------------------------------------------------------------------------------- | ------------------------: |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr> | `6.48 kB`<br />`2.203 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/bdbf3ae799e368080db7af64f385f5c35c4c8fdf'>bdbf3ae799e368080db7af64f385f5c35c4c8fdf</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: apps/public-docsite-v9/src/Concepts/AdvancedConfiguration.stories.mdx [hunk 1]
```diff
@@ -114,6 +114,23 @@ function MyComponent(props) {
 
 Having multiple applications using `FluentProvider` on a single web page can cause interop problems when they come from different bundles, more details in [microsoft/fluentui#26496](https://github.com/microsoft/fluentui/pull/26496).
 
+### React 18
+
+React 18 provides new [`createRoot`](https://react.dev/reference/react-dom/client/createRoot#parameters) API that allows to configure `identifierPrefix` on it:
```
Comment (ling1726) [inline apps/public-docsite-v9/src/Concepts/AdvancedConfiguration.stories.mdx:119]:
```suggestion
It's possible to handle id prefixing natively with React 18 using the [`createRoot`](https://react.dev/reference/react-dom/client/createRoot#parameters) API by configuring `identifierPrefix` on it:
```

**Review by ling1726 — APPROVED** [review]

**Review by ling1726 — APPROVED** [review]

**Pull request merged**